sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/SkipExternalizeTestRecording.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation given to some tests to skip externalizing recording to enable running test validations locally.
```

Comment:
I am not sure we need this annotation. Validating locally or using externalized recordings should be similar and shouldn't require tests to have this annotation.

Reply:
I was thinking if it might be needed when you are performing local validations similar to this test [here](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-test/src/test/java/com/azure/core/test/TestProxyTests.java#L271) reading local session-record files we should at least have an option.

Reply:
I think we should add special handling to `azure-core-test` for this alone rather than something generalized as other SDKs shouldn't be doing something like that.